src/Nethermind/Nethermind.Merge.AuRa/AuRaMergePlugin.cs

```diff
@@ -73,10 +87,101 @@ protected override PostMergeBlockProducerFactory CreateBlockProducerFactory()
+
+                _shutterP2P = new(shutterTxSource.OnDecryptionKeysReceived, _auraConfig, _api.LogManager);
+                _shutterP2P.Start(_auraConfig.ShutterKeyperP2PAddresses);
+            }
+
+            return _api.BlockProducerEnvFactory.Create(shutterTxSource);
+        }
+
         public bool ShouldRunSteps(INethermindApi api)
         {
             _mergeConfig = api.Config<IMergeConfig>();
             return _mergeConfig.Enabled && api.ChainSpec.SealEngineType == SealEngineType.AuRa;
         }
+
+        public new void DisposeAsync()
+        {
+            _shutterP2P?.DisposeAsync();
+            _ = base.DisposeAsync();
+        }
+
+        private void ValidateShutterConfig(IAuraConfig auraConfig)
+        {
+            if (auraConfig.ShutterSequencerContractAddress is null || !Address.TryParse(auraConfig.ShutterSequencerContractAddress, out _))
+            {
+                throw new ArgumentException("Must set Shutter sequencer contract address to valid address.");
+            }
+
+            if (auraConfig.ShutterValidatorRegistryContractAddress is null || !Address.TryParse(auraConfig.ShutterValidatorRegistryContractAddress, out _))
+            {
+                throw new ArgumentException("Must set Shutter validator registry contract address to valid address.");
+            }
+
+            if (auraConfig.ShutterKeyBroadcastContractAddress is null || !Address.TryParse(auraConfig.ShutterKeyBroadcastContractAddress, out _))
+            {
+                throw new ArgumentException("Must set Shutter key broadcast contract address to valid address.");
+            }
+
+            if (auraConfig.ShutterKeyperSetManagerContractAddress is null || !Address.TryParse(auraConfig.ShutterKeyperSetManagerContractAddress, out _))
+            {
+                throw new ArgumentException("Must set Shutter keyper set manager contract address to valid address.");
+            }
+
+            foreach (string addr in auraConfig.ShutterKeyperP2PAddresses)
+            {
+                try
+                {
+                    Multiaddress.Decode(addr);
+                }
+                catch (NotSupportedException)
+                {
+                    throw new ArgumentException($"Could not decode Shutter keyper p2p address \"{addr}\".");
+                }
+            }
+        }
+
+        private Dictionary<ulong, byte[]> LoadValidatorInfo(string fp)
+        {
+            string validatorsInfoRaw = File.ReadAllText(fp);
+            Dictionary<ulong, string>? validatorsInfoParsed = JsonSerializer.Deserialize<Dictionary<ulong, string>>(JsonDocument.Parse(validatorsInfoRaw));
+            if (validatorsInfoParsed is null)
+            {
+                throw new JsonException("Invalid JSON document format, should be (index, public key) pairs.");
+            }
+            return validatorsInfoParsed.ToDictionary(x => x.Key, x => Convert.FromHexString(x.Value.AsSpan()[2..]));
```

Comment:
This parsing is not great:
- loads all file instead of using stream-like access
- does double conversion, first to string then to byte[]